### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by mikolaslorant**
## Summary

* added tests for user and book controllers.
* added tests for user and book daos.

## Trello 
https://trello.com/c/Omz3TgHv/37-test-controllers-y-daos

### CODE REVIEW

File: training/src/test/java/wolox/training/controllers/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        mvc.perform(get("/api/books/")
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", hasSize(2)))
+            .andExpect(jsonPath("$[0].title", is(oneTestBook.getTitle())))
+            .andExpect(jsonPath("$[1].title", is(twoTestBook.getTitle())));
+    }
+
+    @Test
+    public void whenDeleteBook_thenBookNotFound() throws Exception {
```
Comment (leandrocostawolox) [inline training/src/test/java/wolox/training/controllers/BookControllerTest.java]: Maybe you can have this test as whenDeleteBookWhichNotExists_thenBookNotFound()
And another one like whenDeleteBookWhichExists_thenOkResponse

File: training/src/test/java/wolox/training/controllers/UserControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        mvc.perform(get("/api/users/")
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", hasSize(2)))
+            .andExpect(jsonPath("$[0].username", is(oneTestUser.getUsername())))
+            .andExpect(jsonPath("$[1].username", is(twoTestUser.getUsername())));
+    }
+
+    @Test
+    public void whenDeleteUser_thenUserNotFound() throws Exception {
```
Comment (leandrocostawolox) [inline training/src/test/java/wolox/training/controllers/UserControllerTest.java]: Same comment than before here :)

File: training/src/test/java/wolox/training/models/BookTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    public void whenCreateBookWithoutUsername_thenThrowException() {
+        oneTestBook.setTitle(null);
+        bookRepository.save(oneTestBook);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenCreateBookWithEmptyTitle_thenThrowException() {
+        oneTestBook.setTitle("");
+        bookRepository.save(oneTestBook);
+    }
```
Comment (leandrocostawolox) [inline training/src/test/java/wolox/training/models/BookTest.java:72]: Ideally you should test this kind of behaviour with each required field, it's to have more test coverage
In this case is just a training so don't worry about it, I just wanted to tell you so you can have it in mind :)

**Review by leandrocostawolox — COMMENTED** [review]
Good! I wrote some things

File: training/src/test/java/wolox/training/models/BookTest.java (hunk 3 above)
Comment (mikolaslorant) [reply training/src/test/java/wolox/training/models/BookTest.java:72]: ok :D

File: training/src/test/java/wolox/training/controllers/UserControllerTest.java (hunk 2 above)
Comment (mikolaslorant) [reply training/src/test/java/wolox/training/controllers/UserControllerTest.java]: done

File: training/src/test/java/wolox/training/controllers/BookControllerTest.java (hunk 1 above)
Comment (mikolaslorant) [reply training/src/test/java/wolox/training/controllers/BookControllerTest.java]: done

**Review by IgnOliveto — APPROVED** [review]

**Review by leandrocostawolox — APPROVED** [review]

**Pull request merged**